ecs_deploy/ecs.py

```diff
@@ -473,6 +595,267 @@ def apply_container_secrets(self, container, new_secrets, exclusive=False):
         container[u'secrets'] = [
             {"name": s, "valueFrom": merged[s]} for s in merged
         ]
+    
+    def set_system_controls(self, system_controls_list, exclusive=False):
+        system_controls = {}
+        for system_control in system_controls_list:
+            system_controls.setdefault(system_control[0], [])
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

You are using the `dict.setdefault()` method. We recommend using the `defaultdict` object to handle missing items in internal state because the default value is automatically generated when you access or modify a missing key. `defaultdict` can also be easier to use and understand.

[Learn more](https://docs.python.org/3/library/collections.html#collections.defaultdict)

Reply:
Should be fine now.